hips/hip-00xx-sequencing.md

```diff
@@ -0,0 +1,213 @@
+```
+In this example, Helm would be responsible for resolving the annotations on these three resources and deploy all resources in the following order. Resources in `database` and `queue` layers would be deployed at the same time. They would need to be ready before attempting to deploy `app` layer resources:
+
+```
+"database" layer:    [db-service]
+                          ||
+"queue" layer:            || [queue-processor]
+                          ||       ||
+                          ||       ||
+                          \/       \/
+                           \\     //
+                            \\   //
+                             \/ \/
+"app" layer:                 [my-app]
+```
+
+#### Chart dependencies example
+
+```yaml
+name: foo
+annotations:
+  helm.sh/depends-on/subcharts: ["bar", "rabbitmq"]
+dependencies:
+  - name: nginx
+    version: "18.3.1"
+    repository: "oci://registry-1.docker.io/bitnamicharts"
+  - name: rabbitmq
+    version: "9.3.1"
+    repository: "oci://registry-1.docker.io/bitnamicharts"
+  - name: bar   # This is a subchart packaged with "foo" in charts dir. It's not pulled from a remote location
+    version: "0.1.0"
+    depends-on: ["nginx", "rabbitmq"]
+    condition: bar.enabled
+```
+
+```
+ [nginx] [rabbitmq]
+    ||       ||
+    ||       ||
+    \/       \/
+     \\     //
+      \/   \/
+       [bar]
+        ||
+        ||
+        \/
+       [foo]
+```
+
+In this example, helm will first install and wait for all resources of `nginx` and `rabbitmq` dependencies to be "ready" before attempting to install `bar` resources. Once all resources of `bar` are "ready" then and only then will `foo` chart resources be installed. `foo` would require `rabbitmq` to be ready but since the subchart resources would have been installed before `bar`, this requirement would have been fulfilled.
+
+This approach of building a directed acyclic graph (DAG) is prone to circular dependencies. During the templating phase, Helm will have logic to detect, and report any circular dependencies found in the chart templates. Helm will also provide a command to print the DAG for development and troubleshooting purposes.
+
+### Readiness
+
+In order to enforce sequencing, Helm will check that resources are ready using `kstatus` [ready condition](https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/kstatus/README.md#the-ready-condition) allowing helm to proceed installing the next group of resources, or fail the install. Users can override checks using optional `helm.sh/readiness-failure` and `helm.sh/readiness-success` annotations which are jsonpath queries to check status fields of the annotated resource. If any of the checks in the lists evaluate as true, the readiness check is determined as successful/failed. `helm.sh/readiness-timeout` annotation can be used to override how long Helm should wait when performing readiness checks before bailing out. It defaults to `10s`.
```

Comment:
Does there need to be some commentary/discussion regarding how to construct `readiness-success` and/or `readiness-failure`?

Reply:
> Does there need to be some commentary/discussion regarding how to construct `readiness-success` and/or `readiness-failure`?

@sabre1041 I update the doc to better explain how readiness checks work. https://github.com/helm/community/pull/373/commits/707dd264f246fc19c7c66518cbf8331128d5aa65